### PR TITLE
Add connect functionality to yarpview

### PR DESCRIPTION
**Pull request by aerydna**
Added connection via gui.

Connecting to empty string will call `yarp::os::Network::disconnect` with the last remote connected via gui.

### CODE REVIEW

File: src/yarpview/src/main.cpp [hunk 1]
```diff
@@ -115,6 +139,19 @@ int main(int argc, char *argv[])
...
     }
 
+    PortConnector            connector;
+    yarp::os::Value          val;
+    yarp::os::ResourceFinder opt;
+    QObject::connect(topLevel, SIGNAL(connectToPort(QString, QString)), &connector, SLOT(connectPort(QString,QString)));
+
+    if (opt.configure(argc, argv))
+    {
+        if(opt.check("PortName")) connector.portName = opt.find("PortName").asString();
```
Comment (Nicogene) [inline src/yarpview/src/main.cpp]: 💀

File: src/yarpview/src/main.cpp [hunk 2]
```diff
@@ -115,6 +139,19 @@ int main(int argc, char *argv[])
...
+        else if(opt.check("name")) connector.portName = opt.find("name").asString();
```
Comment (Nicogene) [inline src/yarpview/src/main.cpp]: 💀

File: src/yarpview/src/main.cpp [hunk 3]
```diff
@@ -115,6 +139,19 @@ int main(int argc, char *argv[])
...
+        else connector.portName = "/yarpview/img:i";
```
Comment (Nicogene) [inline src/yarpview/src/main.cpp]: 💀

File: src/yarpview/src/main.cpp (hunk 1 above)
Comment (Nicogene) [inline src/yarpview/src/main.cpp]: Usually the name of the parameter starts with lower case, so i'd prefer `portName`

**Comment by traversaro** [comment]
It would be good to update the `doc/cmd_yarpview.dox` file, in particular including the details included in the PR description.

File: src/yarpview/src/main.cpp [hunk 4]
```diff
@@ -21,13 +21,37 @@
...
+{
+    Q_OBJECT
+public:
+    std::string portName;
+    std::string remote;
+
+public slots:
+    bool connectPort(QString source, QString carrier)
+    {
+        if (source.toStdString().empty()) yarp::os::Network::disconnect(remote, portName);
```
Comment (barbalberto) [inline src/yarpview/src/main.cpp]: :skull: :skull:

**Comment by barbalberto** [comment]
Can you post a screenshot?

File: src/yarpview/src/main.cpp (hunk 1 above)
Comment (aerydna) [reply src/yarpview/src/main.cpp]: it's not a parameter name we can change.. it is not a new parameter. it was already checked and used inside. i ve just pre-checked it here. There is so much space for refactoring here, expecially for the parameters parsing process but for the moment i made the less invasive change. leaving the refactoring for the future (it was not the pourpose of this pr)

**Comment by aerydna** [comment]
yarpview

File: doc/release/v2_3_72.md [hunk 5]
```diff
@@ -109,6 +109,10 @@ New Features
 * Added `yarp name list` and `yarp clean` actions after importing
   profiling capabilities of `yarpviz`.
 
+#### `yarpview`
+
+* Added in-gui connect functionality (remote port name textfield, carrier textfield and "connect" button)
```
Comment (traversaro) [inline doc/release/v2_3_72.md]: Missing period at the end of the sentence.

File: doc/cmd_yarpview.dox [hunk 6]
```diff
@@ -49,6 +49,8 @@ restore only the original aspect ratio (Image->Original aspect ratio).
 
 From the menu "File" you can also save individual or sequences of frames.
 
+You can connect to remote image source via gui. by remote name is not provided a disonnection command (from the last remote connected via gui) will be called
```
Comment (traversaro) [inline doc/cmd_yarpview.dox]: The period is missing at the end of the sentence. The phrase "by remote name is not provided a disonnection command (from the last remote connected via gui) will be called" is not really clear.

**Pull request closed without merging**